tiledb/platform/cert_file.h

```diff
@@ -129,8 +144,7 @@ class CertFileTraits<LinuxOS> {
       };
 
       for (const std::string& cert : cert_files) {
-        std::error_code ec;
-        if (std::filesystem::exists(cert, ec)) {
```

Comment:
Do you know whether `if constexpr (platform::is_os_linux)` would work here, or does the "marked unavailable" error happen too early in the compilation process for that conditional to apply?

Reply:
It happens to early: https://gist.github.com/davisp/5ef5d9146b51edc336da20960083d1a3

Reply:
Also of note, `-mmacosx-version-min=10.5` doesn't fix it. Locally at least I had to go all the way to `-mmacosx-version-min=11` for it to compile.